packages/replay/src/upload.ts

```diff
@@ -1,20 +1,26 @@
-const crypto = require("crypto");
-const ProtocolClient = require("./client");
-const { defer, maybeLog, isValidUUID } = require("./utils");
-const { sanitize: sanitizeMetadata } = require("../metadata");
-
-let gClient;
-let gClientReady = defer();
-
-async function initConnection(server, accessToken, verbose, agent) {
+import crypto from "crypto";
+import ProtocolClient from "./client";
+import { defer, maybeLog, isValidUUID } from "./utils";
+import { sanitize as sanitizeMetadata } from "../metadata";
+import { Options, RecordingMetadata, SourceMapsEntry } from "./types";
+
+let gClient: ProtocolClient | undefined;
+let gClientReady = defer<boolean>();
+
+async function initConnection(
+  server: string,
+  accessToken?: string,
+  verbose?: boolean,
+  agent?: any
+) {
   if (!gClient) {
     let { resolve } = gClientReady;
     gClient = new ProtocolClient(
       server,
       {
         async onOpen() {
           try {
-            await gClient.setAccessToken(accessToken);
+            await gClient?.setAccessToken(accessToken);
```

Comment:
:hand: We should be pretty sure it exists on this line, right? Looks like `!` would be more appropriate here